scripts/lib/install

```diff
@@ -218,7 +224,7 @@ case " $os_id $os_id_like " in
     *' rhel '*)
         if ! yum install -y \
             puppet git curl wget jq \
-            python python3 python-six python3-six crudini \
+            python python3 python-six python3-six crudini certbot \
```

Comment:
Let's just do `apt install certbot` inside `setup-certbot`; we don't want a `Certbot` package on all Puppet roles.

Reply:
We also need to make sure `certbot` gets installed on upgrade if `certbot-auto` was in use.

Reply:
Yeah, I'd recommend just adding `certbot` to the `nginx.pp` packages in puppet for that.

We just also need to install it directly in the setup script because we need it before puppet is run.

Reply:
Why *do* we use it before Puppet is run? Wouldn’t it make more sense to have a Puppet class for `certbot` that configures everything inside Puppet?

Reply:
@timabbott @andersk  Should we install `certbot` in `setup-certbot` and then have a check in `certbot-maybe-review ` script or we are going for the puppet class for `certbot`?

Reply:
Hmm, good question.  My guess is that it's a bit of a chicken-and-egg problem.  To use certbot in the standalone mode, we need nginx to not be running when we run certbot, which is kinda a messy thing to setup in puppet.  Or to use it in webroot mode, we need nginx running, but it won't start until a certificate is provided.  

(And I think we didn't have proper support for the `webroot` mode when we first added the tool)

What we don't want is `nginx` to fail to start because the certificate isn't available yet.

So there's probably a world where we make the installer:
(1) The --certbot option doesn't run certbot, it just sets `zulip.conf` to do so in puppet and then sets up the self-signed cert
(2) Then puppet manages the dance of running certbot in webroot mode

But that would probably end up being worse; `puppet` doesn't have the best error messages, so it's probably better to have `certbot` run from shell anyway for handling users providing a hostname they don't control (or whatever).

Reply:
Pushed the changes. Is this what we want?

Reply:
Yes, except I don't see `Package[certbot]` having been added to `nginx.pp`?

Reply:
@timabbott Do we want something else? I have already di something like [this](https://github.com/zulip/zulip/pull/14571/files/32a6583baa49672c017b8e63c3f9ecfdeec2035c#r432221898)